mlflow/metrics/metric_definitions.py

```diff
@@ -249,3 +244,95 @@ def _rougeLsum_eval_fn(eval_df, metrics):
             scores=scores,
             aggregate_results=standard_aggregations(scores),
         )
+
+
+def _mae_eval_fn(predictions, targets, metrics, sample_weight=None):
+    if targets is not None and len(targets) != 0:
+        from sklearn.metrics import mean_absolute_error
+
+        mae = mean_absolute_error(targets, predictions, sample_weight=sample_weight)
+        return MetricValue(aggregate_results={"mean_absolute_error": mae})
```

Comment:
What happens in cases where targets is None? Do we want to throw a warning or something that MAE was not computed? Thoughts? 
cc @dbczumar  @annzhang-db 

Note this implies to many of the metrics

Reply:
I believe this issue was discussed on Ann's previous PR too , that's the reason why I kept the behavior like this https://github.com/mlflow/mlflow/pull/9515#discussion_r1318961931